### PR TITLE
Commit during guest reboot

### PR DESCRIPTION
**Pull request by aliang123**
blockdev_commit_reboot:commit during guest reboot

Signed-off-by: Aihua Liang <aliang@redhat.com>

id: 1820491

### CODE REVIEW

**Comment by aliang123** [comment]
@ngu-niny, @zhencliu, please help to review, thanks

File: qemu/tests/cfg/blockdev_commit_reboot.cfg [hunk 1]
```diff
@@ -0,0 +1,34 @@
+- blockdev_commit_reboot:
+    type = blockdev_commit_reboot
+    virt_test_type = qemu
+    only Linux
+    start_vm = yes
+    kill_vm = yes
+    storage_pools = default
+    storage_type_default = "directory"
+    storage_pool = default
+    image_size_data = 500M
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_reboot.cfg:10]: a data disk is required for this case ? I think we just make snapshots for system image, right?

File: qemu/tests/cfg/blockdev_commit_reboot.cfg [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    image_name_data = data
+    snapshot_tags = sn1 sn2 sn3 sn4
+
+    image_size_sn1 = 20G
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_reboot.cfg:14]: we can use system image's size for all snapshots, so no need to define snx's size here

File: qemu/tests/cfg/blockdev_commit_reboot.cfg [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    image_name_sn1 = sn1
+    image_format_sn1 = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_reboot.cfg:16]: the format can be any besides qcow2?

File: qemu/tests/blockdev_commit_reboot.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from provider import backup_utils
+from provider import job_utils
+
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitReboot(BlockDevCommitTest):
+
+    def configure_data_disk(self, tag):
+        os_type = self.params["os_type"]
```
Comment (zhencliu) [inline qemu/tests/blockdev_commit_reboot.py:12]: No need to get path, all you want is a dir like /tmp for linux, or C: for windows, so, just fill disks_info with that information only. 
It's not a good idea to rewrite configure_data_disk here, for what you want is  the system image, so why not write a method in base class?

File: qemu/tests/cfg/blockdev_commit_reboot.cfg [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    image_name_sn4 = sn4
+    image_size_sn4 = 20G
+    image_format_sn4 = qcow2
+
+    device_tag = "image1"
+    format = qcow2
+    rebase_mode = unsafe
+    Host_RHEL.m8:
+        qemu_force_use_drive_expression = no
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_reboot.cfg:34]: you can see that there are so much duplicated informat in cfg files, such as sn1/2/3/4, is it possible to define one cfg file like blockdev_commit.cfg, then put the reboot/commit as its variants? At that time, you might get a clean cfg.

File: qemu/tests/cfg/blockdev_commit_reboot.cfg (hunk 1 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg]: Have updated, thanks for pointing it out.

File: qemu/tests/cfg/blockdev_commit_reboot.cfg (hunk 5 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg:32]: Yes, but was suggested one case, one cfg is better.

File: qemu/tests/cfg/blockdev_commit_reboot.cfg (hunk 3 above)
Comment (xutian) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg:14]: We only support qcow2 format snapshot.

Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg:14]: yes, as xutian explained, if the base is raw/qcow2, we just use qcow2 as snapshot format.

File: qemu/tests/cfg/blockdev_commit_reboot.cfg (hunk 2 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg]: Have updated, thanks for pointing it out.

File: qemu/tests/blockdev_commit_reboot.py (hunk 4 above)
Comment (aliang123) [reply qemu/tests/blockdev_commit_reboot.py:12]: Have update， thanks

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit_reboot.q35: PASS (101.84 s)

File: provider/blockdev_commit_base.py [hunk 6]
```diff
@@ -94,6 +94,15 @@ def get_linux_disk_path(session, disk_size):
                 return kname
         return None
 
+    def configure_disk(self, tag):
+        if len(self.params["images"].split()) == 1:
```
Comment (zhencliu) [inline provider/blockdev_commit_base.py:98]: what about 'device_tag = image1 data', in this situation, configure_data_disk will be called, but for image1, which is the system image, configure_system_disk should be called.
If there is only one system image in your tests, we can simply check if tag is (self.params['images'].split())[0], if not, i.e. there are multiple system images, we have to find out the list of all images and check if tag is in the list

File: qemu/tests/cfg/blockdev_commit_reboot.cfg [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    image_format_sn2 = qcow2
+
+    image_name_sn3 = sn3
+    image_format_sn3 = qcow2
+
+    image_name_sn4 = sn4
+    image_format_sn4 = qcow2
+
+    device_tag = "image1"
+    format = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_reboot.cfg:25]: what's 'format' used for?

File: provider/blockdev_commit_base.py (hunk 6 above)
Comment (aliang123) [reply provider/blockdev_commit_base.py]: OK, repace it by:
   if tag == "image1":

File: qemu/tests/cfg/blockdev_commit_reboot.cfg (hunk 7 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_reboot.cfg]: Have updated it, thanks

**Comment by zhencliu** [comment]
LGTM

**Comment by YongxueHong** [comment]
Please add a commit for ` blockdev_commit_base.py` since the part of modifying codes is not to new a case.
Thanks.

**Comment by aliang123** [comment]
@yhong, have updated the commit, please help to check. thanks

**Comment by YongxueHong** [comment]
@aliang123  I mean that add a new commit log for that.
```
git add blockdev_commit_base.py
git commit -s
```
And this commit log should befre the case commit log.
Thanks.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit_reboot.q35: PASS (100.85 s)

**Comment by ngu-niny** [comment]
Test PASS:
01-repeat1.Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.blockdev_commit_reboot

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by aliang123** [comment]
Thanks, @ngu-niny, @zhencliu and @YongxueHong

**Pull request merged**